src/adaptation/stepsize.jl

```diff
@@ -17,22 +17,34 @@ end
 
 function DAState(ϵ::AbstractVector{T}) where {T}
     n = length(ϵ)
-    μ = computeμ(ϵ)
+    μ = map(computeμ, ϵ)
     return DAState(0, ϵ, μ, zeros(T, n), zeros(T, n))
 end
 
 function reset!(das::DAState{T}) where {T<:AbstractFloat}
     das.m = 0
     das.μ = computeμ(das.ϵ)
     das.x_bar = zero(T)
-    return das.H_bar = zero(T)
+    das.H_bar = zero(T)
+    return das
 end
 
 function reset!(das::DAState{<:AbstractVector{T}}) where {T<:AbstractFloat}
     das.m = 0
-    das.μ .= computeμ(das.ϵ)
-    das.x_bar .= zero(T)
-    return das.H_bar .= zero(T)
+    map!(computeμ, das.μ, das.ϵ)
```

Comment:
Let's keep this as-is for now. We could refactor the vectorised HMC interface, but better to do it seprately in a concerted effort: 
```suggestion
    das.μ .= computeμ(das.ϵ)
```

Reply:
This suggestion would go against the main intention of the PR, reducing unnecessary allocations: With `map!` (or `das.μ .= computeμ.(das.ϵ)`, but the broadcasting is more stressful for the compiler) no intermediate array would be created in this line, whereas with the suggestion on the right-hand side a new array is allocated that is then copied to `das.μ` (as a side remark, for the compiler `copyto!` should be simpler than broadcasting).